File: c++/Solution242.cpp
//O(n) O(Z), Z = 26
class Solution242 {
public:
    bool isAnagram(string s, string t) {
        //Anagram=字謎, 易位構詞
        if(s.length() != t.length()) return false;
        int count[26] = {0};
        for(int i=0; i<s.length(); ++i)
            ++count[s[i] - 'a'];

        for(int i=0; i<t.length(); ++i){
            if(--count[t[i] - 'a'] < 0)
                return false;
        }
        return true;
    }
};

File: python/Solution242.py
# O(n) O(Z), Z = 26, n = len(s) = len(t)
class Solution242:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        count = [0] * 26
        for c in s:
            count[ord(c) - ord("a")] += 1
            
        for c in t:
            count[ord(c) - ord("a")] -= 1
            if count[ord(c) - ord("a")] < 0:
                return False

        return True
            
        